genienlp/tasks/almond/__init__.py

```diff
@@ -31,86 +31,83 @@
 import torch
 import logging
 from tqdm import tqdm
+from collections import defaultdict
 
 from ..base_task import BaseTask
 from ..registry import register_task
-from .. import generic_dataset
+from ..generic_dataset import CQA
 from ...data_utils.example import Example
 
+from ..base_dataset import Split
+
 logger = logging.getLogger(__name__)
 
 
-class AlmondDataset(generic_dataset.CQA):
+class AlmondDataset(CQA):
     """Obtaining dataset for Almond semantic parsing task"""
 
     base_url = None
 
-    def __init__(self, path, *, make_example, subsample=None, cached_path=None, skip_cache=False,
-                 **kwargs):
+
+    def __init__(self, path, *, make_example, subsample=None, cached_path=None, skip_cache=False, examples=None, **kwargs):
+        
+        if examples:
+            super().__init__(examples, **kwargs)
+            return
```

Comment:
What's this code about? Where are you calling AlmondDataset with the example list? This looks very ugly.

Reply:
it is called here:
`splits[field] = AlmondDataset('', make_example=None, examples=all_examples)`
getting rid of it depends on the resolution of your next comment.